test/declarative_test_fwk/helper.py

```diff
@@ -0,0 +1,251 @@
+        if not resource:
+            logging.warning(f"ResourceReference {reference} not found. Not invoking k8s delete api.")
+            return
+
+        k8s.delete_custom_resource(reference)
+        sleep(self.DEFAULT_WAIT_SECS)
+        self.wait_for_delete(reference) # throws exception if wait fails
+
+    def assert_expectations(self, verb: str, input_data: dict, expectations: dict, reference: k8s.CustomResourceReference) -> None:
+        """Asserts custom resource reference inside Kubernetes cluster against the supplied expectations
+
+        :param verb: expectations after performing the verb (apply, patch, delete)
+        :param input_data: input data to verb
+        :param expectations: expectations to assert
+        :param reference: custom resource reference
+        :return: None
+        """
+        self._assert_conditions(expectations, reference, wait=False)
+        # conditions expectations met, now check current resource against expectations
+        resource = k8s.get_resource(reference)
+        self.assert_items(expectations.get("status"), resource.get("status"))
+
+        # self._assert_state(expectations.get("spec"), resource)  # uncomment to support spec assertions
+
+    def wait_for(self, wait_expectations: dict, reference: k8s.CustomResourceReference) -> None:
+        """Waits for custom resource reference details inside Kubernetes cluster to match supplied config,
+        currently supports wait on "status.conditions",
+        it can be enhanced later for wait on any/other properties.
+
+        Args:
+            wait_expectations: properties to wait for
+            reference:  custom resource reference
+
+        Returns:
+            None
+        """
+
+        # wait for conditions
+        self._assert_conditions(wait_expectations, reference, wait=True)
+
+    def _assert_conditions(self, expectations: dict, reference: k8s.CustomResourceReference, wait: bool = True) -> None:
+        expect_conditions: dict = {}
+        if "status" in expectations and "conditions" in expectations["status"]:
+            expect_conditions = expectations["status"]["conditions"]
+
+        for (condition_name, expected_value) in expect_conditions.items():
+            if type(expected_value) is str:
+                # Example: ACK.Terminal: "True"
+                if wait:
+                    assert k8s.wait_on_condition(reference, condition_name, expected_value, wait_periods=30)
+                else:
+                    actual_condition = k8s.get_resource_condition(reference, condition_name)
+                    assert actual_condition is not None
+                    assert expected_value == actual_condition.get("status")
+
+            elif type(expected_value) is dict:
+                # Example:
+                # ACK.ResourceSynced:
+                #     status: "False"
+                #     message: "Expected message ..."
+                #     timeout: 30 # minutes
```

Comment:
Seconds, not minutes :)

Reply:
Updated code to consider the value in seconds.